controllers/snapshot/snapshot_adapter.go

```diff
@@ -235,6 +234,9 @@ func (a *Adapter) EnsureStaticIntegrationPipelineRunsExist() (controller.Operati
 			} else {
 				pipelineRun, err := a.createIntegrationPipelineRun(a.application, &integrationTestScenario, a.snapshot)
 				if err != nil {
+					if clienterrors.IsInvalid(err) {
+						return a.HandlePipelineCreationError(err, &integrationTestScenario, testStatuses)
+					}
 					a.logger.Error(err, "Failed to create pipelineRun for snapshot and scenario")
 					return controller.RequeueWithError(err)
```

Comment:
```suggestion
					return a.HandlePipelineCreationError(err, &integrationTestScenario, testStatuses)
```
I think it can be simplified since `HandlePipelineCreationError` is doing the same thing.
Other than this, LGTM

Reply:
Thanks for catching that, meant to update that along with the first call and missed it

Reply:
Thanks!